portal/src/graphql/adminapi/UserDetailsAccountSecurity.tsx

```diff
@@ -1,9 +1,227 @@
-import React from "react";
+import React, { useMemo, useCallback, useContext } from "react";
+import cn from "classnames";
+import { List, PrimaryButton } from "@fluentui/react";
+import { FormattedMessage, Context } from "@oursky/react-messageformat";
+
+import ListCellLayout from "../../ListCellLayout";
+import { formatDatetime } from "../../util/formatDatetime";
 
 import styles from "./UserDetailsAccountSecurity.module.scss";
 
-const UserDetailsAccountSecurity: React.FC = function UserDetailsAccountSecurity() {
-  return <div className={styles.root}></div>;
+interface Authenticator {
+  id: string;
+  type: string;
+  is_secondary?: boolean;
+  claims: Record<string, unknown>;
+  createdAt: string;
+  updatedAt: string;
+}
+
+interface UserDetailsAccountSecurityProps {
+  authenticators: Authenticator[];
+}
+
+interface AuthenticatorListItem {
+  label: React.ReactNode;
+  description: React.ReactNode;
+  descriptionClassName?: string;
+  onDetailButtonClick?: () => void;
+}
+
+// authenticator type recognized by portal
+enum AuthenticatorType {
+  Password = "password",
+  OneTimePassword = "oob_otp",
+  TimeBasedOneTimePassword = "totp",
+}
```

Comment:
Prefer union type.

Reply:
AuthenticatorType is just a enum of possible `type` in authenticator while `type` has type `string`